src/main/webapp/app/lecture/lecture-attachments.component.ts

```diff
@@ -46,30 +51,31 @@ export class LectureAttachmentsComponent implements OnInit, OnDestroy {
     private dialogErrorSource = new Subject<string>();
     dialogError$ = this.dialogErrorSource.asObservable();
 
-    constructor(
-        protected activatedRoute: ActivatedRoute,
-        private attachmentService: AttachmentService,
-        private lectureService: LectureService,
-        private fileService: FileService,
-    ) {}
-
-    ngOnInit() {
-        this.notificationText = undefined;
-        this.activatedRoute.parent!.data.subscribe(({ lecture }) => {
-            if (this.lectureId) {
-                this.lectureService.findWithDetails(this.lectureId).subscribe((lectureResponse: HttpResponse<Lecture>) => {
-                    this.lecture = lectureResponse.body!;
-                    this.loadAttachments();
+    private routeDataSubscription?: Subscription;
+
+    constructor() {
+        effect(
+            () => {
+                this.notificationText = undefined;
+                this.routeDataSubscription?.unsubscribe(); // in case the subscription was already defined
```

Comment:
How could it be defined before the constructor?

Reply:
right now it is within an effect, so if `lectureId()` would change, we would overwrite the previous subscription and loose track if the pervious one (potentially causing memory leaks)

It _should_ never happen though